src/mainContent/deviceManagement/components/deviceFiltersComponent.js

```diff
@@ -16,11 +17,33 @@ const DeviceFiltersComponent = (props) => {
         state: []
     };
 
-    let serialNumber = deviceManagementUtility.payload.serialNumber;
+     let serialNumber = deviceManagementUtility.payload.searchPattern;
 
     Object.keys(filters).map(function (filter) {
         props.filtersData[filter].map((item) => {
-            filters[filter].push({ value: item })
+            let selectedStatus = false;
+            if (filter == 'deviceType') {
+                deviceManagementUtility.payload['deviceTypes'].map((persistItem) => {
+                    if (persistItem == item) { selectedStatus = true };
+                });
+            } else if (filter == 'fwUpgradeStatus') {
+                deviceManagementUtility.payload['fwUpgradeStatuses'].map((persistItem) => {
+                    if (persistItem == item) { selectedStatus = true };
+                });
+            } else if (filter == 'profileStatus') {
+                deviceManagementUtility.payload['profileStatuses'].map((persistItem) => {
+                    if (persistItem == item) { selectedStatus = true };
+                });
+            } else if (filter == 'state') {
+                deviceManagementUtility.payload['states'].map((persistItem) => {
+                    if (persistItem == item) { selectedStatus = true };
+                });
+            } else {
+                deviceManagementUtility.payload[filter].map((persistItem) => {
+                    if (persistItem == item) { selectedStatus = true };
+                });
+            }
```

Comment:
I am not sure why you need this extra code to map the keys, instead you can use the keys with same name itself, let me know if you have any specific reason.